splinepy/microstructure/microstructure.py

```diff
@@ -342,160 +299,374 @@ def create(self, closing_face=None, knot_span_wise=None, **kwargs):
                     new_knots = []
                     for i, nks in enumerate(n_k_span):
                         new_knots.extend(
-                            np.linspace(
-                                ukvs[i_pd][i], ukvs[i_pd][i + 1], nks + 2
-                            )[1:-1]
+                            np.linspace(ukv[i], ukv[i + 1], nks + 2)[1:-1]
                         )
-                    deformation_function_copy.insert_knots(i_pd, new_knots)
+                    additional_knots.append(new_knots)
 
-        # Bezier Extraction for composition
-        def_fun_patches = deformation_function_copy.extract.beziers()
+        # Return all knots
+        return additional_knots
 
-        # Calculate parametric space representation for parametrized
-        # microstructures
-        is_parametrized = self.parametrization_function is not None
+    def _get_auxiliary_information(
```

Comment:
aux info of/for what? Can this be a "stand-alone" (instead of member) function?

Reply:
you can always access member variables in python. Anyways, I'd just prefer a bit more informative function name than "aux"

Reply:
I changed the name to `_compute_tiling_prerequisites` as it precomputes all information related to the splitting and bezier-extraction process that is necessary to achieve the correct tiling....